internal/client/ethereum_client.go

```diff
@@ -24,12 +24,44 @@ type NewHeadHandler struct {
 	OnError   func(failure string)
 }
 
+type Client ethclient.Client
+
 //go:generate mockery --output ../mocks --name EthClient --with-expecter
 type EthClient interface {
 	SubscribeNewHead(ctx context.Context, ch chan<- *types.Header) (ethereum.Subscription, error)
 	HeaderByNumber(ctx context.Context, number *big.Int) (*types.Header, error)
 	PeerCount(ctx context.Context) (uint64, error)
 	SyncProgress(ctx context.Context) (*ethereum.SyncProgress, error)
+	Latency(ctx context.Context, method string) (time.Duration, error)
+}
+
+func (c *Client) SubscribeNewHead(ctx context.Context, ch chan<- *types.Header) (ethereum.Subscription, error) {
+	return (*ethclient.Client)(c).SubscribeNewHead(ctx, ch)
+}
+
+func (c *Client) HeaderByNumber(ctx context.Context, number *big.Int) (*types.Header, error) {
+	return (*ethclient.Client)(c).HeaderByNumber(ctx, number)
+}
+
+func (c *Client) PeerCount(ctx context.Context) (uint64, error) {
+	return (*ethclient.Client)(c).PeerCount(ctx)
+}
+
+func (c *Client) SyncProgress(ctx context.Context) (*ethereum.SyncProgress, error) {
+	return (*ethclient.Client)(c).SyncProgress(ctx)
+}
+
+// Latency calls the specified RPC method using the given context and returns the duration of the call,
+// as well as the error if one occurred. No arguments are passed to the RPC method.
+//
+// TODO(polsar): If the method expects one or more arguments, it will return an error that will be passed
+// to the caller. We should detect this type of error and not return it, since the call has otherwise succeeded,
+// which is the only thing the caller cares about.
+func (c *Client) Latency(ctx context.Context, method string) (time.Duration, error) {
```

Comment:
nit: Rename to `RecordLatency` or something? The other methods here are names of actual RPC methods, where as `Latency` is not.